kv/kvserver: TestStoreRangeMergeSlowUnabandonedFollower_WithSplit failed

kv/kvserver.TestStoreRangeMergeSlowUnabandonedFollower_WithSplit [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3916406&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3916406&tab=artifacts#/) on master @ [f6f60c6cd9da2300540cda93422aad3e033880ed](https://github.com/cockroachdb/cockroach/commits/f6f60c6cd9da2300540cda93422aad3e033880ed):


```
=== RUN   TestStoreRangeMergeSlowUnabandonedFollower_WithSplit
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/751d67000aac5f3394c2369309253f02/logTestStoreRangeMergeSlowUnabandonedFollower_WithSplit2913701310
    test_log_scope.go:80: use -show-logs to present logs inline
    client_merge_test.go:2774: condition failed to evaluate within 45s: rhs store does not own valid lease for rhs range
        goroutine 8287559 [running]:
        runtime/debug.Stack()
        	GOROOT/src/runtime/debug/stack.go:24 +0x65
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithin({0x56e9958, 0xc017722000}, 0xb, 0x0)
        	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:60 +0x5f
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon({0x56e9958, 0xc017722000}, 0x2)
        	github.com/cockroachdb/cockroach/pkg/testutils/soon.go:41 +0x4a
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestStoreRangeMergeSlowUnabandonedFollower_WithSplit(0xc017722000)
        	github.com/cockroachdb/cockroach/pkg/kv/kvserver_test/client_merge_test.go:2774 +0x889
        testing.tRunner(0xc017722000, 0x4807740)
        	GOROOT/src/testing/testing.go:1259 +0x102
        created by testing.(*T).Run
        	GOROOT/src/testing/testing.go:1306 +0x35a
    panic.go:642: -- test log scope end --
--- FAIL: TestStoreRangeMergeSlowUnabandonedFollower_WithSplit (46.48s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStoreRangeMergeSlowUnabandonedFollower_WithSplit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestStoreRangeMergeSlowUnabandonedFollower_WithSplit [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3919887&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3919887&tab=artifacts#/) on master @ [45390e7b59618d508a0075bae99bb015e61805b2](https://github.com/cockroachdb/cockroach/commits/45390e7b59618d508a0075bae99bb015e61805b2):


```
=== RUN   TestStoreRangeMergeSlowUnabandonedFollower_WithSplit
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestStoreRangeMergeSlowUnabandonedFollower_WithSplit3829434542
    test_log_scope.go:80: use -show-logs to present logs inline
    client_merge_test.go:2774: condition failed to evaluate within 45s: rhs store does not own valid lease for rhs range
        goroutine 193597 [running]:
        runtime/debug.Stack()
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x65
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithin({0x569de38, 0xc005c38d00}, 0xb, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:60 +0x5f
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon({0x569de38, 0xc005c38d00}, 0x2)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:41 +0x4a
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestStoreRangeMergeSlowUnabandonedFollower_WithSplit(0xc005c38d00)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_merge_test.go:2774 +0x889
        testing.tRunner(0xc005c38d00, 0x47bc670)
        	/usr/local/go/src/testing/testing.go:1259 +0x102
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1306 +0x35a
    panic.go:642: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestStoreRangeMergeSlowUnabandonedFollower_WithSplit3829434542
--- FAIL: TestStoreRangeMergeSlowUnabandonedFollower_WithSplit (46.82s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStoreRangeMergeSlowUnabandonedFollower_WithSplit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestStoreRangeMergeSlowUnabandonedFollower_WithSplit [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3936331&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3936331&tab=artifacts#/) on master @ [3b4e180a23f5121e0d4106eb3dc5f61ebc314188](https://github.com/cockroachdb/cockroach/commits/3b4e180a23f5121e0d4106eb3dc5f61ebc314188):


```
=== RUN   TestStoreRangeMergeSlowUnabandonedFollower_WithSplit
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestStoreRangeMergeSlowUnabandonedFollower_WithSplit4081635205
    test_log_scope.go:80: use -show-logs to present logs inline
    client_merge_test.go:2776: condition failed to evaluate within 45s: rhs store does not own valid lease for rhs range
        goroutine 162546 [running]:
        runtime/debug.Stack()
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x65
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithin({0x56bd878, 0xc0066df040}, 0xb, 0x0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:60 +0x5f
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon({0x56bd878, 0xc0066df040}, 0x2)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:41 +0x4a
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestStoreRangeMergeSlowUnabandonedFollower_WithSplit(0xc0066df040)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_merge_test.go:2776 +0x889
        testing.tRunner(0xc0066df040, 0x47c7f40)
        	/usr/local/go/src/testing/testing.go:1259 +0x102
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1306 +0x35a
    panic.go:642: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestStoreRangeMergeSlowUnabandonedFollower_WithSplit4081635205
--- FAIL: TestStoreRangeMergeSlowUnabandonedFollower_WithSplit (45.87s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStoreRangeMergeSlowUnabandonedFollower_WithSplit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestStoreRangeMergeSlowUnabandonedFollower_WithSplit [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3962387&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3962387&tab=artifacts#/) on master @ [1903b8f7e8195d92cd6ddb281b7fed764900f5da](https://github.com/cockroachdb/cockroach/commits/1903b8f7e8195d92cd6ddb281b7fed764900f5da):


```
=== RUN   TestStoreRangeMergeSlowUnabandonedFollower_WithSplit
    test_log_scope.go:79: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestStoreRangeMergeSlowUnabandonedFollower_WithSplit706578554
    test_log_scope.go:80: use -show-logs to present logs inline
    client_merge_test.go:2776: condition failed to evaluate within 3m45s: rhs store does not own valid lease for rhs range
        goroutine 169366 [running]:
        runtime/debug.Stack()
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x72
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithin({0x95e9818, 0xc006322680}, 0x49961c, 0xc005c677a0)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:60 +0x7d
        github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon({0x95e9818, 0xc006322680}, 0x2)
        	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:41 +0x5b
        github.com/cockroachdb/cockroach/pkg/kv/kvserver_test.TestStoreRangeMergeSlowUnabandonedFollower_WithSplit(0xc006322680)
        	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvserver/client_merge_test.go:2776 +0xc33
        testing.tRunner(0xc006322680, 0x66266e8)
        	/usr/local/go/src/testing/testing.go:1259 +0x230
        created by testing.(*T).Run
        	/usr/local/go/src/testing/testing.go:1306 +0x727
    panic.go:661: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestStoreRangeMergeSlowUnabandonedFollower_WithSplit706578554
--- FAIL: TestStoreRangeMergeSlowUnabandonedFollower_WithSplit (230.78s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStoreRangeMergeSlowUnabandonedFollower_WithSplit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/client_merge_test.go
+++ b/pkg/kv/kvserver/client_merge_test.go
@@ -2756,7 +2756,7 @@ func TestStoreRangeMergeSlowUnabandonedFollower_WithSplit(t *testing.T) {
 		t.Fatal(pErr)
 	}
 
-	// Now split the newly merged range splits back out at exactly the same key.
+	// Now split the newly merged range back out at exactly the same key.
 	// When the replica GC queue looks in meta2 it will find the new RHS range, of
 	// which store2 is a member. Note that store2 does not yet have an initialized
 	// replica for this range, since it would intersect with the old RHS replica.
@@ -2769,7 +2769,7 @@ func TestStoreRangeMergeSlowUnabandonedFollower_WithSplit(t *testing.T) {
 	tc.RemoveVotersOrFatal(t, lhsDesc.StartKey.AsRawKey(), tc.Target(2))
 
 	// Transfer the lease on the new RHS to store2 and wait for it to apply. This
-	// will force its replica to of the new RHS to become up to date, which
+	// will force its replica of the new RHS to become up to date, which
 	// indirectly tests that the replica GC queue cleans up both the LHS replica
 	// and the old RHS replica.
 	tc.TransferRangeLeaseOrFatal(t, *newRHSDesc, tc.Target(2))

--- a/pkg/kv/kvserver/split_trigger_helper.go
+++ b/pkg/kv/kvserver/split_trigger_helper.go
@@ -37,7 +37,7 @@ func (rd *replicaMsgAppDropper) ShouldDrop(startKey roachpb.RKey) (fmt.Stringer,
 	if lhsRepl == nil {
 		return nil, false
 	}
-	lhsRepl.store.gcQueue.AddAsync(context.Background(), lhsRepl, replicaGCPriorityDefault)
+	lhsRepl.store.replicaGCQueue.AddAsync(context.Background(), lhsRepl, replicaGCPriorityDefault)
 	return lhsRepl, true
 }
 
@@ -48,7 +48,7 @@ type msgAppDropper interface {
 
 // maybeDropMsgApp returns true if the incoming Raft message should be dropped.
 // It does so if the recipient replica is uninitialized (i.e. has no state) and
-// is waiting for a split trigger to apply,in which case  delivering the message
+// is waiting for a split trigger to apply,in which case delivering the message
 // in this situation would result in an unnecessary Raft snapshot: the MsgApp
 // would be rejected and the rejection would prompt the leader to send a
 // snapshot, while the split trigger would likely populate the replica "for

